### PR TITLE
Stabilize Lucene indexing

### PR DESCRIPTION
**Pull request by aprattes**
This commit stabilizes Lucene indexing by:
* preventing the Lucene scheduler thread from dying and never starting
  again
* continuing indexing even when an error occurs on one or more repos
* removing the explicit call to Java GC (`System.gc()`) in the Lucene
  service

### CODE REVIEW

File: src/main/java/com/gitblit/service/LuceneService.java [hunk 1]
```diff
@@ -160,33 +160,40 @@ public LuceneService(
...
+					if (model.hasCommits && !ArrayUtils.isEmpty(model.indexedBranches)) {
+						Repository repository = repositoryManager.getRepository(model.name);
+						if (repository == null) {
+							if (repositoryManager.isCollectingGarbage(model.name)) {
+								logger.info(MessageFormat.format("Skipping Lucene index of {0}, busy garbage collecting", repositoryName));
+							}
+							continue;
+						}
+						index(model, repository);
+						repository.close();
```
Comment (ederst) [inline src/main/java/com/gitblit/service/LuceneService.java]: Please coloe `repository` in a final block.

File: src/main/java/com/gitblit/service/LuceneService.java [hunk 2]
```diff
@@ -160,33 +160,40 @@ public LuceneService(
...
-					continue;
+				} catch(Exception e) {
+					logger.error("Error while indexing lucene on repository " + repositoryName, e);
 				}
-				index(model, repository);
-				repository.close();
-				System.gc();
 			}
+		} catch (Exception e) {
+			logger.error("Error while indexing lucene!", e);
```
Comment (ederst) [inline src/main/java/com/gitblit/service/LuceneService.java]: Maybe better: "General error while indexing lucene:" (use ":" or no punctuation)

File: src/main/java/com/gitblit/service/LuceneService.java [hunk 3]
```diff
@@ -160,33 +160,40 @@ public LuceneService(
...
+								logger.info(MessageFormat.format("Skipping Lucene index of {0}, busy garbage collecting", repositoryName));
+							}
+							continue;
+						}
+						index(model, repository);
+						repository.close();
 					}
-					continue;
+				} catch(Exception e) {
+					logger.error("Error while indexing lucene on repository " + repositoryName, e);
```
Comment (ederst) [inline src/main/java/com/gitblit/service/LuceneService.java]: Maybe better: "Error while indexing lucene for repository"

**Review by ederst — CHANGES_REQUESTED** [review]

**Comment by aprattes** [comment]
Hi @ederst !
I have updated my commit... please review.

**Comment by aprattes** [comment]
@ederst,looks great!

**Review by aprattes — COMMENTED** [review]
looks great!

**Review by ederst — APPROVED** [review]

**Pull request merged**